utils/check_dist_content.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+import sys
+from pathlib import Path
+from tarfile import TarFile
+from zipfile import ZipFile
+
+dist_path = Path("dist")
+wheel_path = next(dist_path.glob("*.whl"))
+sdist_path = next(dist_path.glob("*.tar.gz"))
+
+with ZipFile(wheel_path) as wheel_file:
+    # Allow only 'narwhals' and 'narwhals-<version>.dist-info' (metadata)
+    unexpected_wheel_dirs = {
+        dir_name
+        for name in wheel_file.namelist()
+        if not (dir_name := name.split("/")[0]).startswith("narwhals")
+    }
+
+    if unexpected_wheel_dirs:
+        print(f"🚨 Unexpected directories in wheel: {unexpected_wheel_dirs}")  # noqa: T201
+        sys.exit(1)
+
+with TarFile.open(sdist_path, mode="r:gz") as sdist_file:
+    # Allow only 'narwhals' ans 'tests' folders
+    sdist_dirs = {m.name.split("/")[1] for m in sdist_file.getmembers()}
+    allowed_sdist_dirs = {
+        "narwhals",
+        "tests",
+        "pyproject.toml",
+        "PKG-INFO",
+        "LICENSE.md",
+        "README.md",
+        ".gitignore",
```

Comment:
Are these expected? Apparently 

```toml
[tool.hatch.build.targets.sdist]
include = [
  "narwhals/*",
  "tests/*",
]
```
will include them

Reply:
Mmm I think so (based on also what polars does)

Reply:
Yes. Hatchling always includes them (they are in current sdists too), and well, at least first four are expected. `.gitignore` is kinda debatable, but apparently Hatchling uses it sometimes to determine what to install ([not necessarily in a beneficial way](https://github.com/pypa/hatch/issues/1964)).